File: src/main/java/me/sul/abnormalstate/AbnormalState.java
package me.sul.abnormalstate;

import me.sul.abnormalstate.bleeding.Bleeding;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

public final class AbnormalState extends JavaPlugin {

    @Override
    public void onEnable() {
        Bukkit.getPluginManager().registerEvents(new Bleeding(this), this);
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }
}

File: src/main/java/me/sul/abnormalstate/AbnormalState.kt
package me.sul.abnormalstate

import me.sul.abnormalstate.bleeding.Bleeding
import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin
import org.bukkit.plugin.java.JavaPlugin

class AbnormalState : JavaPlugin() {
    companion object {
        lateinit var instance : Plugin
    }

    override fun onEnable() {
        instance = this
        Bukkit.getPluginManager().registerEvents(Bleeding, this)
    }

    override fun onDisable() {
    }
}